### ISSUE TITLE
TypeError exception from Python binding

### ISSUE THREAD

**Issue by dongahn** [flux-framework/flux-core#1532]
I am probably doing something incorrect but I was getting an TypeError exception from Python binding.

```
Exception TypeError: 'isinstance() arg 2 must be a class, type, or tuple
of classes and types' in <bound method Flux.__del__ of
<flux.core.handle.Flux object at 0x2aaab0339750>> ignored
```

You should be able to reproduce this on quartz using this PR: https://github.com/flux-framework/flux-workflow-examples/pull/3

```
quartz386{dahn}188: salloc -N 3 -ppdebug
quartz30{dahn}6: srun --pty --mpi=none -N3 /usr/global/tools/flux/toss_3_x86_64_ib/default/bin/flux start -o,-S,log-filename=out
quartz30{dahn}7: bookkeeper.py 5
bookkeeper: all jobs submited
bookkeeper: waiting until all jobs complete
bookkeeper: all jobs completed
    ID NTASKS STATE                    START      RUNTIME    RANKS COMMAND
     1      6 exited     2018-05-18T18:48:01       5.724s    [0-2] compute.py
     2      3 exited     2018-05-18T18:48:01       5.742s    [0-2] io-forwarding
     3      6 exited     2018-05-18T18:48:01       5.729s    [0-2] compute.py
     4      3 exited     2018-05-18T18:48:01       5.647s    [0-2] io-forwarding
     5      6 exited     2018-05-18T18:48:01       5.661s    [0-2] compute.py
     6      3 exited     2018-05-18T18:48:01       5.611s    [0-2] io-forwarding
     7      6 exited     2018-05-18T18:48:01       5.667s    [0-2] compute.py
     8      3 exited     2018-05-18T18:48:01       5.633s    [0-2] io-forwarding
     9      6 exited     2018-05-18T18:48:01       5.583s    [0-2] compute.py
    10      3 exited     2018-05-18T18:48:01       5.608s    [0-2] io-forwarding
Exception TypeError: 'isinstance() arg 2 must be a class, type, or tuple of classes and types' in <bound method Flux.__del__ of <flux.core.handle.Flux object at 0x2aaab0339750>> ignored

**Comment by SteVwonder** [comment]
Is this still a problem or was this resolved by #1537?

**Comment by dongahn** [comment]
I believe this has been resolved. Thanks.
